Gdax/IProductClient.cs

```diff
@@ -26,7 +26,7 @@ public interface IProductClient
 		/// <param name="end">End time in ISO 8601.</param>
 		/// <param name="granularity">Desired timeslice in seconds.</param>
 		/// <returns></returns>
-		Task<IList<HistoricRate>> GetHistoricRates(String productId, DateTime? start = null, DateTime? end = null, Int32? granularity = null);
+		Task<IList<HistoricRate>> GetHistoricRates(String productId, DateTime? start = null, DateTime? end = null, MarketPeriod? granularity = null);
```

Comment:
I think it would make more sense for this to take a TimeSpan so that consumers can choose exactly what the granularity is.